utils/cosmos/iscn/sign.ts

```diff
@@ -25,31 +26,63 @@ export function formatISCNTxPayload(payload: ISCNRegisterPayload): ISCNSignPaylo
     authorNames,
     authorUrls,
     authorWallets,
+    likerIds,
+    descriptions,
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     cosmosWallet,
     ...data
   } = payload;
+
   const contentFingerprints = []
   if (fileSHA256) contentFingerprints.push(`hash://sha256/${fileSHA256}`)
   if (ipfsHash) contentFingerprints.push(`ipfs://${ipfsHash}`)
-  const stakeholders = []
+  const stakeholders: any = []
   if (authorNames.length) {
     for (let i = 0; i < authorNames.length; i += 1) {
-      const authorName = authorNames[i]
-      const authorUrl = authorUrls[i]
-      const authorId = authorWallets[i]
-      const isNonEmpty = authorUrl || authorName || authorId
+      const authorName: string = authorNames[i]
+      const sameAs: any = []
+      const identifier: any = []
+      const description = descriptions[i]
+      const url: string = likerIds[i]
+        ? `https://like.co/${likerIds[i]}`
+        : authorUrls[i][0] || authorName
+
+      authorWallets[i].forEach((a: any) => {
+        if (a.type === 'cosmos') {
+          identifier.push({
+            '@type': 'PropertyValue',
+            propertyID: WALLET_TYPE_REPLACER[a.type],
+            value: `did:cosmos:${a.address.slice(6)}`,
+          })
+        } else {
+          identifier.push({
+            '@type': 'PropertyValue',
+            propertyID: WALLET_TYPE_REPLACER[a.type],
+            value: `did:${a.type}:${a.address}`,
+          })
+        }
+      })
+      if (authorUrls[i].length) {
+        authorUrls[i].forEach((a: any) => {
+          if (a.length) {
+            sameAs.push(a)
+          }
+        })
+      }
```

Comment:
No need to check for `.length`, since if you are sure `authorUrls` is array (since you called `.length`), a forEach on empty array would not cause any issue.

Why need to check for `a.length`? What error case are we guarding?

Reply:
本來會一直 push 一個 empty string 上去 sameAs..... 結果剛剛重新 yarn dav 就沒問題了